pkg/networkserver/grpc_gsns.go

```diff
@@ -996,19 +997,24 @@ func (ns *NetworkServer) handleDataUplink(ctx context.Context, up *ttnpb.UplinkM
 		log.FromContext(ctx).WithError(err).Error("Failed to update downlink task queue after data uplink")
 	}
 	if !matched.IsRetransmission {
+		var airtimep *time.Duration
+		if airtime, err := toa.Compute(len(up.RawPayload), up.Settings); err == nil {
+			airtimep = &airtime
+		}
```

Comment:
```suggestion
		var airtime *time.Duration
		t, err := toa.Compute(len(up.RawPayload), up.Settings)
		if err != nil {
			log.FromContext(ctx).WithError(err).Debug("Failed to compute time-on-air")
		} else {
			airtime = &t
		}
```

It `toa.Compute` expected to fail? If so, when?

Reply:
Invalid parameters. Probably not in this context.